### PR TITLE
PERF-3746 Rewrite BackgroundIndexConstruction workload

### PR DESCRIPTION
**Pull request by jordiolivares**
This PR rewrites the BackgroundIndexConstruction workload to reflect a more realistic scenario.

It lowers the dataset size so that we can repeat index creation to reduce noise and overall runs with better timings.

It is a clean cut from the previous workload, only the Setup actor has been kept with the same name.

### CODE REVIEW

**Comment by jordiolivares** [comment]
@brad-devlugt Added you in because it's somewhat related to @haleyConnelly's previous PR (https://github.com/mongodb/genny/pull/799). Feel free to swap places with someone else in the Performance team if necessary.

**Review by brad-devlugt — APPROVED** [review]
Thanks @jordiolivares  for the new workload.  Looks good to me with some patch builds to check for noise levels. @haleyConnelly did a really great job on this in https://github.com/mongodb/genny/pull/799 and can likely help you out with getting that set up. Feel free to reach out with additional questions!

File: src/workloads/execution/BackgroundIndexConstruction.yml [hunk 1]
```diff
@@ -19,31 +19,70 @@ Keywords:
...
     Threads: 1
-    CollectionCount: 1
+    CollectionCount: &CollCount 1
     DocumentCount: 10_000_000
     BatchSize: 1000
-    Document:
-      randomInt: {^RandomInt: {min: -10000000, max: 10000000}}
-      randomString: {^RandomString: {length: 16}}
+    Document: *SmallDoc
   - &Nop {Nop: true}
```
[the remark this answers is not in the document]
Comment (jordiolivares) [reply src/workloads/execution/BackgroundIndexConstruction.yml]: I can make most of them use it, but the IndexCollection actor has different config depending on the phase. I can split it into two actors, but the improvement in readability is dubious at best.

File: src/workloads/execution/BackgroundIndexConstruction.yml [hunk 2]
```diff
@@ -19,31 +19,70 @@ Keywords:
...
-- Name: InsertData
+- Name: LoadSmallData
   Type: Loader
   Threads: 1
   Phases:
   - Repeat: 1
     Database: &db test
     Threads: 1
-    CollectionCount: 1
+    CollectionCount: &CollCount 1
```
[the remark this answers is not in the document]
Comment (jordiolivares) [reply src/workloads/execution/BackgroundIndexConstruction.yml]: That was a leftover from previous experiments, thanks for the catch!

File: src/workloads/execution/BackgroundIndexConstruction.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (haleyConnelly) [reply src/workloads/execution/BackgroundIndexConstruction.yml]: Its fine then, easier not to bother if it complicates things too much

**Pull request merged**